From 05b5befcbac2d907e6a7b317895bfbbeef74443d Mon Sep 17 00:00:00 2001
From: Peter Volkov <pva@gentoo.org>
Date: Thu, 22 Jul 2010 18:30:38 +0400
Subject: [PATCH] vzifup-post: fix device for proxy arp

We need to del/add routing to detect correct device for proxy arp in
NEIGHBOUR_DEVS=detect mode.

http://bugzilla.openvz.org/show_bug.cgi?id=1587
---
 bin/vzifup-post.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/bin/vzifup-post.in b/bin/vzifup-post.in
index bba7d11..2979907 100755
--- a/bin/vzifup-post.in
+++ b/bin/vzifup-post.in
@@ -24,7 +24,9 @@ update_arp()
 {
 	local veip
 	for veip in $(awk '!/^Version/ { print $1 }' /proc/vz/veip); do
+		[ "$NEIGHBOUR_DEVS" != 'detect' ] && vzdelrouting "${veip}"
 		vzarp add ${veip}
+		[ "$NEIGHBOUR_DEVS" != 'detect' ] && vzaddrouting "${veip}"
 	done
 }
 
-- 
1.7.1

